python/lsst/dax/distribution/DataGenServer.py

```diff
@@ -0,0 +1,513 @@
+#!/usr/bin/env python3
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import getopt
+import itertools
+import os
+import socket
+import sys
+import threading
+import yaml
+
+from enum import Enum
+
+from DataGenConnection import DataGenConnection
+from DataGenConnection import DataGenError
+from DataIngest import DataIngest
+
+class GenerationStage(Enum):
+    """This class is used to indicate where a chunk is in the process
+    of having synthetic data genrated.
+
+    UNASSIGNED : The chunk has not been assigned to a worker to be generated.
+    ASSIGNED : The chunk has been assigend to a worker.
+    FINISHED : The assigned worker has finished generating the chunk.
+    LIMBO : The chunk was assigned but never finished.
+    """
+    UNASSIGNED = 1
+    ASSIGNED = 2
+    FINISHED = 3
+    LIMBO = 4
+
+
+class ChunkInfo:
+    """Information about a chunk including its status
+
+    Parameters
+    ----------
+    chunk_id : int
+        The chunk id number
+
+    Members
+    -------
+    gen_stage : enum class
+        This indactes how far along the chunk is in the generation process.
+    client_id : int
+        The id of the client program generating the chunk.
+    client_addr : string
+        The IP address of the client generating the chunk.
+    """
+
+    def __init__(self, chunk_id):
+        self.chunk_id = chunk_id
+        self.gen_stage = GenerationStage(GenerationStage.UNASSIGNED)
+        self.client_id = '-1'
+        self.client_addr = None
+
```

Comment:
client_addr string type - use "" instead?

Reply:
I like None in that it is obvious that client_addr was never defined instead of it being set by an empty string in log file. If this is a python faux pas, then changing it isn't a big deal. I will add a comment that it should be string.